### PR TITLE
Provision and clean-up all external resources

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

This pull request extends existing cluster GKE integration job for Kyma PRs with provisioning and de-provisioning of all "cluster" resources (IP reservation, DNS entry)
The job is not complete yet (Kyma installation and testing is missing) but it's an important step forward.

Changes proposed in this pull request:

- Create all external resources necessary for the plan (IP, DNS, cluster)
- Clean up all external resources necessary for the plan (cluster, DNS, IP)
- All scripts shellcheck'ed

**Related issue(s)**
See also #20

### CODE REVIEW

File: prow/images/bootstrap/Dockerfile [hunk 1]
```diff
@@ -13,6 +13,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     curl \
     git \
     jq \
+    dnsutils \
```
Comment (Tomasz-Smelcerz-SAP) [inline prow/images/bootstrap/Dockerfile]: Info to the reviewer: Image 0.0.1 re-built with this change.

File: prow/config.yaml [hunk 2]
```diff
@@ -26,13 +26,13 @@ presets:
...
+    env:
+      - name: GOOGLE_APPLICATION_CREDENTIALS
         value: /etc/credentials/sa-gke-kyma-integration/service-account.json
-      - name: GCLOUD_PROJECT_NAME
-        value: kyma-project
-      - name: GCLOUD_COMPUTE_ZONE
-        value: europe-west3-a
+      - name: CLOUDSDK_DNS_ZONE_NAME
+        value: "build-kyma" #GCloud DNS Zone Name (NOT it's DNS name!)
+      - name: SOURCES_DIR
```
Comment (Tomasz-Smelcerz-SAP) [inline prow/config.yaml]: Note to reviewer:  I think this "SOURCES_DIR" is a value common to many jobs. I would like to extract it to a separate preset (to be reusable) but it's not purpose of _this_ PR, so for now I would just leave it here.

File: prow/scripts/cluster-integration/generate-self-signed-cert.sh [hunk 3]
```diff
@@ -22,8 +22,8 @@ openssl req -x509 -nodes -days 5 -newkey rsa:4069 \
                  -keyout "${KEY_PATH}" \
                  -out "${CERT_PATH}"
 
-TLS_CERT=$(< /cert.pem base64 | tr -d '\n')
-TLS_KEY=$(< /key.pem base64 | tr -d '\n')
+TLS_CERT=$(base64 "${SCRIPTS_DIR}"/cert.pem | tr -d '\n')
```
Comment (aszecowka) [inline prow/scripts/cluster-integration/generate-self-signed-cert.sh]: you already have variables that point to this files

File: prow/config.yaml (hunk 2 above)
Comment (aszecowka) [reply prow/config.yaml]: I would rather call it KYMA_PROJECT_DIR

File: prow/scripts/cluster-integration/generate-self-signed-cert.sh (hunk 3 above)
Comment (Tomasz-Smelcerz-SAP) [reply prow/scripts/cluster-integration/generate-self-signed-cert.sh]: Fixed

File: prow/config.yaml (hunk 2 above)
Comment (Tomasz-Smelcerz-SAP) [reply prow/config.yaml]: Renamed

**Pull request merged**